tests/api2/test_audit_sudo.py

```diff
@@ -0,0 +1,254 @@
+        assert accept['submituser'] == self.USER
+        assert accept['command'] == LS_COMMAND
+        assert accept['runuser'] == SUDO_TO_USER
+        assert accept['runargv'].split(',') == ['ls', '/tmp']
+
+    def allowed_some(self):
+        """Some of the sudo commands are allowed"""
+        # First get a baseline # of events
+        count = user_sudo_events(self.USER, True)
+
+        # Generate a sudo command that we ARE allowed perform
+        magic = ''.join(secrets.choice(string.ascii_letters + string.digits) for i in range(20))
+        self.sudo_command(f'echo {magic}')
+        assert count + 1 == wait_for_events(self.USER, count + 1)
+        accept = assert_accept(user_sudo_events(self.USER)[-1])
+        assert accept['submituser'] == self.USER
+        assert accept['command'] == ECHO_COMMAND
+        assert accept['runuser'] == 'root'
+        assert accept['runargv'].split(',') == ['echo', magic]
+
+        # Generate a sudo command that we are NOT allowed perform
+        with pytest.raises(AssertionError):
+            self.sudo_command('ls /etc')
+        # Returned exception depends upon whether passwd or nopasswd
+        assert count + 2 == wait_for_events(self.USER, count + 2)
+        reject = assert_reject(user_sudo_events(self.USER)[-1])
+        assert reject['submituser'] == self.USER
+        assert reject['command'] == LS_COMMAND
+        assert reject['runuser'] == 'root'
+        assert reject['runargv'].split(',') == ['ls', '/etc']
+        assert reject['reason'] == 'command not allowed'
+
+    def allowed_none(self):
+        """None of the sudo commands are allowed"""
+        # First get a baseline # of events
+        count = user_sudo_events(self.USER, True)
+
+        # Now create an event and do some basic checking to ensure it failed
+        with pytest.raises(AssertionError) as ve:
+            self.sudo_command('ls /etc')
+        assert 'is not allowed to execute ' in str(ve), str(ve)
+        assert count + 1 == wait_for_events(self.USER, count + 1)
+        reject = assert_reject(user_sudo_events(self.USER)[-1])
+        assert reject['submituser'] == self.USER
+        assert reject['command'] == LS_COMMAND
+        assert reject['runuser'] == 'root'
+        assert reject['runargv'].split(',') == ['ls', '/etc']
+        assert reject['reason'] == 'command not allowed'
+
+
+class SudoNoPasswd:
+    def sudo_command(self, cmd, runuser=None):
+        command = self.generate_command(cmd, runuser)
+        ssh(command, user=self.USER, password=self.PASSWORD)
+
+
+class SudoPasswd:
+    def sudo_command(self, cmd, runuser=None):
+        command = f'echo {self.PASSWORD} | {self.generate_command(cmd, runuser, self.PASSWORD)}'
+        ssh(command, user=self.USER, password=self.PASSWORD)
+
+
+class TestSudoAllowedAllNoPasswd(SudoTests, SudoNoPasswd):
+
+    USER = 'sudo-allowed-all-nopw-user'
+    PASSWORD = ''.join(secrets.choice(string.ascii_letters + string.digits) for i in range(10))
```

Comment:
We have a function specifically for purposes like this. It's in the crypto function in utils just FYI